presto-spi/src/main/java/com/facebook/presto/spi/function/SqlInvokedFunction.java

```diff
@@ -141,7 +141,14 @@ public SqlFunctionId getFunctionId()
 
     public FunctionImplementationType getFunctionImplementationType()
     {
-        return FunctionImplementationType.SQL;
+        switch (routineCharacteristics.getLanguage()) {
+            case SQL:
+                return FunctionImplementationType.SQL;
+            case JAVA:
+                return FunctionImplementationType.THRIFT;
+            default:
+                throw new UnsupportedOperationException(format("Unsupported language %s", routineCharacteristics.getLanguage()));
+        }
```

Comment:
Using a static map would be easier?

Reply:
I found this easier to read (code is at the same place). If code needs to be duplicated then I think a static map makes more sense. Currently I think this is fine. We have many this type of case statements in the codebase.